### PR TITLE
Removing deprecated variables and methods and making public variables private

### PR DESCRIPTION
**Pull request by valib**
This is the first batch and I will continue.

### CODE REVIEW

**Comment by valib** [comment]
@SubJunk for now I finished. There are few deprecated methods in the code to decide if and how would be removed.

**Comment by SubJunk** [comment]
@valib thanks, I will review it soon :)

File: src/main/java/net/pms/PMS.java [hunk 1]
```diff
@@ -582,26 +565,28 @@ public void configurationChanged(ConfigurationEvent event) {
 		OutputParams outputParams = new OutputParams(configuration);
 
 		// Prevent unwanted GUI buffer artifacts (and runaway timers)
-		outputParams.hidebuffer = true;
+		outputParams.setHidebuffer(true);
```
Comment (SubJunk) [inline src/main/java/net/pms/PMS.java]: @valib Can we revert the changes in this file before this line? It looks like some automatic formatting happened which is fine but it would be easier for me if we did formatting after we are in sync with DMS because it creates more conflicts for me

Comment (SubJunk) [reply src/main/java/net/pms/PMS.java]: If you want I can do that reverting

File: src/main/java/net/pms/io/OutputParams.java [hunk 2]
```diff
@@ -1,20 +1,19 @@
 /*
- * PS3 Media Server, for streaming any medias to your PS3.
- * Copyright (C) 2008  A.Brochard
+ * PS3 Media Server, for streaming any medias to your PS3. Copyright (C) 2008
```
Comment (SubJunk) [inline src/main/java/net/pms/io/OutputParams.java]: Can we revert the changes to this comment?

File: src/main/java/net/pms/io/OutputParams.java [hunk 3]
```diff
@@ -28,103 +27,321 @@
...
-			", shift_scr=" + shift_scr +
-			", sid=" + sid +
-			", stdin=" + stdin +
-			", timeend=" + timeend +
-			", timeseek=" + timeseek +
-			", toFrame=" + toFrame +
-			", waitbeforestart=" + waitbeforestart +
-			", workDir=" + workDir +
-			", env=" + env + "]";
+		return "OutputParams [aid=" + getAid() + ", avidemux=" + isAvidemux() + ", cleanup=" + isCleanup() + ", forceFps=" + getForceFps()
```
Comment (SubJunk) [inline src/main/java/net/pms/io/OutputParams.java]: Seemed better before? More readable

File: src/main/java/net/pms/io/OutputParams.java [hunk 4]
```diff
@@ -28,103 +27,321 @@
...
+	public void setLossyaudio(boolean lossyaudio) {
+		this.lossyaudio = lossyaudio;
+	}
+
+	public boolean isNo_videoencode() {
+		return no_videoencode;
+	}
+
+	public void setNo_videoencode(boolean no_videoencode) {
+		this.no_videoencode = no_videoencode;
```
Comment (SubJunk) [inline src/main/java/net/pms/io/OutputParams.java]: This could be a good opportunity to also rename these private variables to our usual convention like `isNoVideoEncode` or just `noVideoEncode`. We already have that convention in this file (like `forceFps and forceType` just below here) so it would be good to be consistent. Let me know if you want me to make that change, or if you disagree

File: src/test/java/net/pms/formats/FormatsRegressionTest0.java [hunk 5]
```diff
@@ -47,7 +47,7 @@ public void test05() throws Throwable {
 	public void test07() throws Throwable {
 		MKV mKV0 = new MKV();
 		String str1 = mKV0.toString();
-		String[] str_array2 = mKV0.getId();
+		String[] str_array2 = mKV0.getSupportedExtensions();
```
Comment (SubJunk) [inline src/test/java/net/pms/formats/FormatsRegressionTest0.java]: I'm confused about this change, but haven't looked into it so it might make sense. I am just leaving notes :)

File: src/main/java/net/pms/dlna/DLNAMediaSubtitle.java [hunk 6]
```diff
@@ -206,8 +176,10 @@ private void setFileSubsCharacterSet(String forcedLang) {
 						subsCharacterSet = subsCharacterSet.substring(0, subsCharacterSet.lastIndexOf("-"));
 					}
 
+					String lang = null;
 					if (forcedLang == null) { // set the detected language when the language is not specified in the filename
 						lang = match.getLanguage();
+						setLang(lang);
```
[the remark this answers is not in the document]
Comment (valib) [reply src/main/java/net/pms/dlna/DLNAMediaSubtitle.java]: @SubJunk the `lang` value is used for logging and will be null in this stage when `forcedLang <> null`

File: src/main/java/net/pms/dlna/DLNAResource.java [hunk 7]
```diff
@@ -86,197 +80,69 @@
...
 	/**
-	 * @deprecated Use standard getter and setter to access this field.
-	 *
 	 * The last time refresh was called.
 	 */
-	@Deprecated
-	protected long lastRefreshTime;
-
-	@SuppressWarnings("unused")
-	private String lastSearch;
```
[the remark this answers is not in the document]
Comment (valib) [reply src/main/java/net/pms/dlna/DLNAResource.java]: @SubJunk yes it is not used locally.

File: src/main/java/net/pms/dlna/DLNAResource.java [hunk 8]
```diff
@@ -1562,7 +1418,7 @@ protected void resolveOnce() { }
 	 * removed from it (if supported). There are other mechanisms for that e.g.
 	 * {@link #doRefreshChildren()} (see {@link Feed} for an example).
 	 */
-	public synchronized final void syncResolve() {
+	public synchronized void syncResolve() {
```
[the remark this answers is not in the document]
Comment (valib) [reply src/main/java/net/pms/dlna/DLNAResource.java]: @SubJunk because I made deprecated method `resolve()` as private and changed it to `syncResolve()` in each asociated clases. The final cannot be overriden.

File: src/main/java/net/pms/dlna/DLNAResource.java [hunk 9]
```diff
@@ -1578,10 +1434,9 @@ public synchronized final void syncResolve() {
 	}
 
 	/**
-	 * @deprecated Use {@link #syncResolve()} instead
+	 *  Use {@link #syncResolve()} instead
```
[the remark this answers is not in the document]
Comment (valib) [reply src/main/java/net/pms/dlna/DLNAResource.java]: @SubJunk will do it

File: src/main/java/net/pms/dlna/DLNAResource.java (hunk 7 above)
[the remark this answers is not in the document]
Comment (valib) [reply src/main/java/net/pms/dlna/DLNAResource.java]: @SubJunk `lastSearch` was used locally only at line 1283 and had no purpose as you can see in the original code.

File: src/main/java/net/pms/io/OutputParams.java (hunk 3 above)
Comment (valib) [reply src/main/java/net/pms/io/OutputParams.java]: @SubJunk automatic formatting. I will revert it.

File: src/main/java/net/pms/io/OutputParams.java (hunk 4 above)
Comment (valib) [reply src/main/java/net/pms/io/OutputParams.java]: @SubJunk done in 49ab5c1

File: src/test/java/net/pms/formats/FormatsRegressionTest0.java (hunk 5 above)
Comment (valib) [reply src/test/java/net/pms/formats/FormatsRegressionTest0.java]: @SubJunk removed deprecated method `getId()` from `Format`  class which returned `getSupportedExtensions()`

**Comment by SubJunk** [comment]
@valib thanks for the further work on it. It might be good to merge this to the 8.0.0 branch instead of master, what do you think?

**Comment by valib** [comment]
@SubJunk it is up to you where it will be merged but what features do you plan to have in the 8.0.0 branch?

**Comment by SubJunk** [comment]
@valib I described my idea for it here https://github.com/UniversalMediaServer/UniversalMediaServer/issues/1612
so it is about web streaming.
I have been sidetracked lately with trying to merge bugfixes for other things but I don't think it will take long to get 8.0.0 features finished, since you and I already did that GUI change

**Comment by SubJunk** [comment]
With 200+ issues it is easy to forget them! I have been closing a lot lately and will continue to try to do that.
Ok cool, will merge into 8.0.0 when I have a chance, thanks for this

**Comment by valib** [comment]
@SubJunk this is the last time I have taken the effort to clean up the code. If you not rewiew it than this will be your task in the future. This PR is here almost 2 years.

**Comment by SubJunk** [comment]
@valib that is a fair comment. I will make it a priority

File: src/main/java/net/pms/dlna/DLNAResource.java [hunk 10]
```diff
@@ -3248,24 +3117,24 @@ public synchronized InputStream getInputStream(Range range, RendererConfiguratio
 
 		// Pipe transcoding result
 		OutputParams params = new OutputParams(configurationSpecificToRenderer);
-		params.aid = getMediaAudio();
-		params.sid = media_subtitle;
-		params.header = getHeaders();
-		params.mediaRenderer = mediarenderer;
+		params.setAid(getMediaAudio());
+		params.setSid(media_subtitle);
+		params.setHeader(getHeaders());
```
Comment (SubJunk) [inline src/main/java/net/pms/dlna/DLNAResource.java:3122]: The setter and getter don't match - shouldn't it be either `setHeaders` and `getHeaders`, or `setHeader` and `getHeader`?

File: src/main/java/net/pms/util/FileUtil.java [hunk 11]
```diff
@@ -1915,6 +1875,8 @@ public static String renameForSorting(String filename) {
 	}
 
 	/**
+<<<<<<< HEAD
```
Comment (SubJunk) [inline src/main/java/net/pms/util/FileUtil.java:1878]: Leftovers from a merge conflict

File: src/main/java/net/pms/util/FileUtil.java [hunk 12]
```diff
@@ -1924,6 +1886,7 @@ public static BufferedReader bufferedReaderWithCorrectCharset(File file) throws
 	}
 
 	/**
+>>>>>>> refs/remotes/origin/master
```
Comment (SubJunk) [inline src/main/java/net/pms/util/FileUtil.java:1889]: Leftovers from a merge conflict

**Review by SubJunk — CHANGES_REQUESTED** [review]
This is a partial review since I have to stop for the night, I wanted to get these comments published asap

File: src/main/java/net/pms/configuration/PmsConfiguration.java [hunk 13]
```diff
@@ -3699,17 +3609,7 @@ public void setPreventSleep(PreventSleepMode value) {
...
-		if (value == null && configuration.containsKey(KEY_PREVENTS_SLEEP)) {
-			// Backwards compatibility
-			sleepMode = getBoolean(KEY_PREVENTS_SLEEP, true) ? PreventSleepMode.PLAYBACK : PreventSleepMode.NEVER;
-			configuration.clearProperty(KEY_PREVENTS_SLEEP);
-			configuration.setProperty(KEY_PREVENT_SLEEP, sleepMode.getValue());
-		} else if (value != null) {
-			sleepMode = PreventSleepMode.typeOf(value);
-		}
-		return sleepMode != null ? sleepMode : PreventSleepMode.PLAYBACK; // Default
+		return PreventSleepMode.typeOf(getString(KEY_PREVENT_SLEEP, PreventSleepMode.PLAYBACK.getValue()));
```
[the remark this answers is not in the document]
Comment (valib) [reply src/main/java/net/pms/configuration/PmsConfiguration.java:3612]: @SubJunk only` .getValue()` returns the string to meet the `getString()` method. The method `getString()` is not applicable for the arguments (String, PreventSleepMode)

File: src/main/java/net/pms/dlna/DLNAMediaInfo.java [hunk 14]
```diff
@@ -511,9 +431,35 @@ public void putExtra(String key, String value) {
 		extras.put(key, value);
 	}
 
-	@Deprecated
-	public void generateThumbnail(InputFile input, Format ext, int type, Double seekPosition, boolean resume) {
-		generateThumbnail(input, ext, type, seekPosition, resume, null);
+	public String getExtrasAsString() {
```
[the remark this answers is not in the document]
Comment (valib) [reply src/main/java/net/pms/dlna/DLNAMediaInfo.java:434]: @SubJunk you are right this is wrong merging.

File: src/main/java/net/pms/dlna/DLNAResource.java (hunk 10 above)
Comment (valib) [reply src/main/java/net/pms/dlna/DLNAResource.java:3122]: @SubJunk it seems like plugins implementation whose are no longer supported.

Comment (SubJunk) [reply src/main/java/net/pms/dlna/DLNAResource.java]: I think we can do that, but let's do it after this branch merges?

File: src/main/java/net/pms/io/OutputParams.java [hunk 15]
```diff
@@ -28,103 +27,341 @@
...
 	 * Set some values to allow fast streaming start of transcoded videos
 	 */
 	public void manageFastStart() {
-		if (mediaRenderer != null && mediaRenderer.isTranscodeFastStart()) {
-			waitbeforestart = 0; // no delay when the transcode is starting
-			minBufferSize = 1; // 1Mb of minimum buffer before sending the file
+		if (getMediaRenderer() != null && getMediaRenderer().isTranscodeFastStart()) {
+			setWaitBeforeStart(0); // no delay when the transcode is starting
+			setMinBufferSize(1); // 1Mb of minimum buffer before sending the
+								 // file
```
Comment (SubJunk) [inline src/main/java/net/pms/io/OutputParams.java:90]:
```suggestion
			setMinBufferSize(1); // 1Mb of minimum buffer before sending the file
```

File: src/main/java/net/pms/configuration/PmsConfiguration.java (hunk 13 above)
[the remark this answers is not in the document]
Comment (SubJunk) [reply src/main/java/net/pms/configuration/PmsConfiguration.java:3612]: Ok cool. But is the logic the same as before? That's a lot of lines to remove from the function. The preventing sleep feature seems to work well for me so I am worried about breaking it

File: src/main/java/net/pms/dlna/DLNAResource.java (hunk 10 above)
Comment (valib) [reply src/main/java/net/pms/dlna/DLNAResource.java]: @SubJunk because the `getHeaders()` is used only locally and is alvays `null` we can remove it now.

File: src/main/java/net/pms/configuration/PmsConfiguration.java (hunk 13 above)
[the remark this answers is not in the document]
Comment (valib) [reply src/main/java/net/pms/configuration/PmsConfiguration.java]: @SubJunk thanks for the comment. Fixed.

[the remark this answers is not in the document]
Comment (valib) [reply src/main/java/net/pms/configuration/PmsConfiguration.java:3612]: @SubJunk the logic is the same but not keeping backwards compatibility with deprecated `KEY_PREVENTS_SLEEP`

[the remark this answers is not in the document]
Comment (SubJunk) [reply src/main/java/net/pms/configuration/PmsConfiguration.java:3612]: Ok

**Review by SubJunk — APPROVED** [review]

**Comment by SubJunk** [comment]
@valib this seems to have caused a big bug, I suspect the refactoring of `sync` methods but haven't tried reverting them.

The bug is that on a fresh database, certain folders of the Media Library do not work.
Reproduction steps:

1) Delete the database folder
2) Start UMS
3) Wait until it finishes the startup scan
4) Go to `Media Library > Video > Movies` and it should be empty

If you repeat the steps on the last commit before this was merged `605f7c0` they work.

Are you able to work on this soon? If not I can revert the PR until you have time

**Comment by valib** [comment]
@SubJunk I take a look.

**Comment by SubJunk** [comment]
Cool. It happens on existing databases too with new files. I suspect you can probably reproduce it by adding like `Test.Series.S01E01.1080p.mkv` and then going into `Media Library > Video > TV Shows > Test Series` and that folder would be empty
